.maintain/frame-weight-template.hbs

```diff
@@ -0,0 +1,104 @@
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Autogenerated weights for {{pallet}}
+//!
+//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION {{version}}
+//! DATE: {{date}}, STEPS: `{{cmd.steps}}`, REPEAT: {{cmd.repeat}}, LOW RANGE: `{{cmd.lowest_range_values}}`, HIGH RANGE: `{{cmd.highest_range_values}}`
+//! EXECUTION: {{cmd.execution}}, WASM-EXECUTION: {{cmd.wasm_execution}}, CHAIN: {{cmd.chain}}, DB CACHE: {{cmd.db_cache}}
+
+// Executed Command:
+//{{#each args as |arg|}} {{arg}}{{/each}}
+
+#![cfg_attr(rustfmt, rustfmt_skip)]
+#![allow(unused_parens)]
+#![allow(unused_imports)]
+#![allow(clippy::unnecessary_cast)]
+
+use frame_support::{traits::Get, weights::{Weight, constants::RocksDbWeight}};
+use sp_std::marker::PhantomData;
+
+/// Weight functions needed for {{pallet}}.
+pub trait WeightInfo {
+    {{#each benchmarks as |benchmark|}}
+    fn {{benchmark.name~}}
+    (
+        {{~#each benchmark.components as |c| ~}}
+        {{c.name}}: u32, {{/each~}}
+    ) -> Weight;
+    {{/each}}
+}
+
+/// Weights for {{pallet}} using the Substrate node and recommended hardware.
+pub struct SubstrateWeight<T>(PhantomData<T>);
+impl<T: frame_system::Config> {{pallet}}::WeightInfo for SubstrateWeight<T> {
+    {{#each benchmarks as |benchmark|}}
+    fn {{benchmark.name~}}
+    (
+        {{~#each benchmark.components as |c| ~}}
+        {{~#if (not c.is_used)}}_{{/if}}{{c.name}}: u32, {{/each~}}
+    ) -> Weight {
+        ({{underscore benchmark.base_weight}} as Weight)
+            {{#each benchmark.component_weight as |cw|}}
+            // Standard Error: {{underscore cw.error}}
+            .saturating_add(({{underscore cw.slope}} as Weight).saturating_mul({{cw.name}} as Weight))
+            {{/each}}
+            {{#if (ne benchmark.base_reads "0")}}
+            .saturating_add(T::DbWeight::get().reads({{benchmark.base_reads}} as Weight))
+            {{/if}}
+            {{#each benchmark.component_reads as |cr|}}
+            .saturating_add(T::DbWeight::get().reads(({{cr.slope}} as Weight).saturating_mul({{cr.name}} as Weight)))
+            {{/each}}
+            {{#if (ne benchmark.base_writes "0")}}
+            .saturating_add(T::DbWeight::get().writes({{benchmark.base_writes}} as Weight))
+            {{/if}}
+            {{#each benchmark.component_writes as |cw|}}
+            .saturating_add(T::DbWeight::get().writes(({{cw.slope}} as Weight).saturating_mul({{cw.name}} as Weight)))
+            {{/each}}
+    }
+    {{/each}}
+}
+
+// For backwards compatibility and tests
+impl WeightInfo for () {
```

Comment:
This is also unnecessary because of the implementation in `pallets/gear/weights.rs`

Reply:
Common practice to leave it.

Reply:
This template is needed for pallet weights also.